rocketchat-core/src/main/java/com/rocketchat/core/RestImpl.java

```diff
@@ -163,6 +169,85 @@ public void onResponse(Call call, Response response) throws IOException {
         });
     }
 
+    void getRoomFiles(String roomId,
+                      BaseRoom.RoomType roomType,
+                      String offset,
+                      String sortBy,
+                      Sort sort,
+                      final RoomCallback.GetFilesCallback callback) {
+        checkNotNull(offset,"roomId == null");
+        checkNotNull(offset,"roomType == null");
+        checkNotNull(offset,"offset == null");
+        checkNotNull(sortBy,"sortBy == null");
+        checkNotNull(sort,"sortDirection == null");
+        checkNotNull(callback,"callback == null");
+
+        String path;
+        switch (roomType) {
+            case PUBLIC:
+                path = "channels.files";
+                break;
+            case PRIVATE:
+                path = "groups.files";
+                break;
+            default:
+                path = "dm.files";
+                break;
+        }
+
+        RequestBody body = new FormBody.Builder()
+                .add("roomId", roomId)
+                .add("offset", offset)
+                .add("sort", "{\"" + sortBy + "\":" + sort.getDirection() + "}")
+                .build();
+
+        Request request = requestBuilder(path)
+                .post(body)
+                .build();
+
+        client.newCall(request).enqueue(new okhttp3.Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                callback.onError(new RocketChatNetworkErrorException("network error", e));
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if (!response.isSuccessful()) {
+                    processCallbackError(response, callback);
+                    return;
+                }
+
+                try {
+                    JSONObject json = new JSONObject(response.body().string());
+                    System.out.println("RESPONSE: " + json.toString());
+
+                    JSONArray filesJSONArray = json.getJSONArray("files");
+                    int length = filesJSONArray.length();
+                    ArrayList<Attachment> attachments = new ArrayList<>();
```

Comment:
This is not wrong, but the usual preferred method is to declare the variable only with the `List` interface, instead of its concrete type `ArrayList`, also, since you know the size of the array, you could initialize it with a proper initial size:
`List<Attachment> attachments = new ArrayList<>(lenght);`

I'm also planning to generify this handling code, using a generic response Class and moshi to parse it directly.

Reply:
Yes! You are right! I forgot about this simple, but efficient details!